repository/src/lib.rs

```diff
@@ -424,9 +425,112 @@ impl<T: RawRepository> DistributedRepository<T> {
     /// Creates a block commit on top of the `work` branch.
     pub async fn create_block(
         &mut self,
-        _author: PublicKey,
+        author: PublicKey,
     ) -> Result<(BlockHeader, CommitHash), Error> {
-        unimplemented!()
+        let work_commit = self.raw.locate_branch(WORK_BRANCH_NAME.into()).await?;
+        let last_header_commit = self.raw.locate_branch(FINALIZED_BRANCH_NAME.into()).await?;
+
+        // Check if the `work` branch is rebased on top of the `finalized` branch.
+        if self
+            .raw
+            .find_merge_base(last_header_commit, work_commit)
+            .await?
+            != last_header_commit
+        {
+            return Err(anyhow!(
+                "branch {} should be rebased on {}",
+                WORK_BRANCH_NAME,
+                FINALIZED_BRANCH_NAME
+            ));
+        }
+
+        // Check the validity of the commit sequence
+        let commits = read_commits(self, last_header_commit, work_commit).await?;
+        let last_header = self.get_last_finalized_block_header().await?;
+        let mut reserved_state = self.get_reserved_state().await?;
```

Comment:
Please checkout to the target branch (`work` in this case) first before retrieving the RS.

Reply:
understood and fixed